packages/myst-transforms/src/frontmatter.ts

```diff
@@ -34,7 +35,10 @@ export function getFrontmatter(
   tree: GenericParent,
   opts: Options = { propagateTargets: true },
 ): { tree: GenericParent; frontmatter: Record<string, any>; identifiers: string[] } {
-  if (opts.propagateTargets) mystTargetsTransform(tree);
+  if (opts.propagateTargets) {
+    liftMystDirectivesAndRolesTransform(tree);
```

Comment:
(This is where all the prerequisites are called - trying to keep it minimal)

Reply:
Yes - I'd agree with that. There were probably edge cases of, for example, a `(target)=` inside an admonition that would not resolve correctly. Now, `index` directives make this no longer an edge case.